### ISSUE TITLE
[Bug] ouisync-app: Weird behavior when opening a ouisync repo link from within Signal

### ISSUE THREAD

**Issue by Juul** [equalitie/ouisync-app#421]
Originally reported this on the `ouisync` repo but realized that was wrong and closed it, then moved it here.

I'm running ouisync 0.8 from github releases on ungoogled Graphene OS latest stable and Signal beta (not installed via play store) and when I open a ouisync link from within Signal, it opens correctly but in my list of running apps (where you can move between running apps (the swipe-up gesture I think is the default to access it on Android)). ouisync and Signal don't appear separately. This sometimes happens when e.g. viewing a PDF where the app that's opened from within Signal acts more like a preview inside of Signal rather than running fully separately (from a UI perspective). In any case, this wouldn't be a problem except for this:

* I can't get back to Signal without exiting ouisync
* When I exit ouisync by left-swiping from the right side twice (after the first time it says "Press back again to exit") then it drops me back to the Android launcher instead of back to Signal. Signal is still running and I can swipe up and find it in the app chooser, but when I switch back to Signal I still see ouisync and it's still running! I can't get back to Signal's UI.
* When I then kill Signal from the app chooser and restart Signal from the launcher then I get back to Signal's UI
* After killing and re-starting Signal, I then tried starting ouisync normally from the launcher and it had lost the ouisync repo that I had previously completely finished syncing after opening it from within Signal! The repo list was back to being empty.

I should add that when trying to re-import any of the repos that went missing from the list, using the same repo link and name, I get "There is already a repository with this name".

Hm it might be important that I had already started ouisync manually before I clicked that link inside Signal? Not sure. Could this be related to the singleTop launchMode?

**Comment by Juul** [comment]
Yeah it seems like the repos are still there taking up space but I just can't access them :S

**Comment by madadam** [comment]
Thanks for reporting this! This should now be fixed in master and will be part of the next release. Fixed by changing `launchMode` to `singleInstance`.

**Comment by Juul** [comment]
Nice! Any idea how to delete those invisible repos?

**Comment by madadam** [comment]
Good question. I'm afraid there is no way currently, apart from deleting the whole app. But that would delete all repos, not just those orphaned ones. Maybe we can implement a way to detect these orphaned repos on app startup and re-insert them into the repo list so you would be then able to delete them normally (or keep them). We should also probably do something with the underlying race condition that caused this issue in the first place (the repo list being modified concurrently from two app instances).

**Comment by Juul** [comment]
Thanks I'll try that!

**Comment by IvanaBlzvc** [comment]
redmine #27750

**Comment by IvanaBlzvc** [comment]
https://redmine.equalit.ie/issues/27750 - closed - closing this issue
